### PR TITLE
*: remove alter-primary-key configuration

### PR DESCRIPTION
**Pull request by tangenta**
### First-time contributors' checklist 

- [x] I've signed [**Contributor License Agreement**](https://cla-assistant.io/pingcap/docs-cn) that's required for repo owners to accept my contribution.

### What is changed, added or deleted? (Required)

The `alter-primary-key` config option is deprecated & removed in v5.0, this PR update the related documentation.

### Which TiDB version(s) do your changes apply to? (Required)

- [x] master (the latest development version)
- [x] v5.0 (TiDB 5.0 versions)
- [ ] v4.0 (TiDB 4.0 versions)
- [ ] v3.1 (TiDB 3.1 versions)
- [ ] v3.0 (TiDB 3.0 versions)
- [ ] v2.1 (TiDB 2.1 versions)

### What is the related PR or file link(s)?

- This PR is translated from: NA
- Other reference link(s): 
    - TiDB PR: https://github.com/pingcap/tidb/pull/23270
    - Related Docs PR: https://github.com/pingcap/docs-cn/pull/5771

### Do your changes match any of the following descriptions?

- [ ] Delete files
- [ ] Change aliases
- [ ] Need modification after applied to another branch 
- [ ] Might cause conflicts after applied to another branch

### CODE REVIEW

**Comment by tangenta** [comment]
@TomShawn This PR has the same precedence as https://github.com/pingcap/docs-cn/pull/5771. PTAL

File: tidb-configuration-file.md [hunk 1]
```diff
@@ -90,11 +90,11 @@ TiDB 配置文件比命令行参数支持更多的选项。你可以在 [config/
 + 将旧表中的 utf8 字符集当成 utf8mb4的开关。
 + 默认值：true
 
-### `alter-primary-key`
+### `alter-primary-key` <span class="version-mark">从 v5.0.0 版本开始废弃</span>
```
Comment (TomShawn) [inline tidb-configuration-file.md:93]: 这里的"废弃"是说还能使用，但不支持吗？还是说从 5.0 开始就没有这个配置项了呢？

Comment (TomShawn) [inline tidb-configuration-file.md:93]: 在 master 和 v5.0 文档中建议这样描述
```suggestion
### `alter-primary-key`（已废弃）
```

File: tidb-configuration-file.md [hunk 2]
```diff
@@ -90,11 +90,11 @@ TiDB 配置文件比命令行参数支持更多的选项。你可以在 [config/
...
 
 + 用于控制添加或者删除主键功能。
 + 默认值：false
-+ 默认情况下，不支持增删主键。将此变量被设置为 true 后，支持增删主键功能。不过对在此开关开启前已经存在的表，且主键是整型类型时，即使之后开启此开关也不支持对此列表删除主键。
++ 默认情况下，不支持增删主键。将此变量被设置为 true 后，支持增删主键功能。不过对在此开关开启前已经存在的表，且主键是整型类型时，即使之后开启此开关也不支持对此列表删除主键。从 v5.0.0 开始，该配置项不再生效。如果需要增删主键，请使用 `NONCLUSTERED` 代替。要了解关于 `CLUSTERED` 主键的详细信息，请参考[聚簇索引](/clustered-indexes.md)。
```
Comment (TomShawn) [inline tidb-configuration-file.md:97]: 建议该信息在 master 版中删除，在 v5.0 中保留。
```suggestion
+ 默认情况下，不支持增删主键。将此变量被设置为 true 后，支持增删主键功能。不过对在此开关开启前已经存在的表，且主键是整型类型时，即使之后开启此开关也不支持对此列表删除主键。该配置项不再生效。如果需要增删主键，请使用 `NONCLUSTERED` 代替。要了解关于 `CLUSTERED` 主键的详细信息，请参考[聚簇索引](/clustered-indexes.md)。
```

**Comment by TomShawn** [comment]
Rest LGTM

File: tidb-configuration-file.md (hunk 2 above)
Comment (tangenta) [reply tidb-configuration-file.md:97]: 这个要改成 note 的格式吗

Comment (TomShawn) [reply tidb-configuration-file.md:97]: 好的

**Comment by TomShawn** [comment]
@coocood PTAL whether this PR can be merged. Thanks!

**Pull request merged**